src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,408 @@
+        }
+      } else {
+        if (objectNode.path("minInclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("minInclusive").get("dateTime").asText()))
+        }
+      }
+
+      val maxInclusive:Option[Either[Int, String]] = if (objectNode.path("maxInclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxInclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxInclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxInclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxInclusive").get("dateTime").asText()))
+        }
+      }
+
+      val minExclusive:Option[Either[Int, String]] = if (objectNode.path("minExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("minExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("minExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.get("minExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("minExclusive").get("dateTime").asText()))
+        }
+      }
+
+      val maxExclusive:Option[Either[Int, String]] = if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+        if (objectNode.path("maxExclusive").isMissingNode) {
+          None
+        } else {
+          Option(Left(objectNode.get("maxExclusive").asInt()))
+        }
+      } else {
+        if (objectNode.path("maxExclusive").path("dateTime").isMissingNode) {
+          None
+        } else {
+          Option(Right(objectNode.get("maxExclusive").get("dateTime").asText()))
+        }
+      }
+
+     (minInclusive, minExclusive, maxInclusive, maxExclusive) match {
+       case (Some(Left(minI)), Some(Left(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (Some(Right(minI)), Some(Right(minE)), _, _) => throw new MetadataError(s"datatype cannot specify both minimum/minInclusive ($minI) and minExclusive ($minE)")
+       case (_, _, Some(Left(maxI)), Some(Left(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (_, _, Some(Right(maxI)), Some(Right(maxE))) => throw new MetadataError(s"datatype cannot specify both maximum/maxInclusive ($maxI) and maxExclusive ($maxE)")
+       case (Some(Left(minI)), _,Some(Left(maxI)), _) if minI > maxI => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
+       case (Some(Right(minI)), _,Some(Right(maxI)), _) if minI.asInstanceOf[Int] > maxI.asInstanceOf[Int] => throw new MetadataError(s"datatype minInclusive ($minI) cannot be more than maxInclusive ($maxI)")
```

Comment:
you don't need to do `minI.asInstanceOf[Int]`, it should already reasily it's an int because it's from the `Right`.